ARCHITECTURE.md

```diff
@@ -0,0 +1,137 @@
+- Defines the `information_schema` table, which is a special table available in all databases and contains some data about the schemas of other tables.
+
+### `sql/analyzer`
+
+The analyzer is perhaps the more complex component of the project. It contains a main component, which is the `Analyzer`, in charge of executing its registered rules on execution trees for resolving some parts, removing redundant data, optimizing things for performance, etc.
+
+There are several phases on the analyzer, because some rules need to be run before others, some need to be executed several times, other just once, etc.
+Inside `rules.go` are all the default rules and the phases in which they're executed.
+
+On top of that, all available rules are defined in this package. Each rule has a specific role in the analyzer. Rules should be as small and atomic as possible and try to do only one job and always produce a tree that is as resolved as the one it received or more.
+
+### `sql/expression`
+
+This package includes the implementation of all the SQL expressions available in go-mysql-server, except functions. Arithmetic operators, logic operators, conversions, etc are implemented here.
+
+Inside `registry.go` there is a registry of all the default functions, even if they're not defined here.
+
+`Inspect` and `Walk` utility functions are provided to inspect expressions.
+
+### `sql/expression/function`
+
+Implementation of all the functions available in go-mysql-server.
+
+### `sql/expression/function/aggregation`
+
+Implementation of all the aggregation functions available in go-mysql-server.
+
+### `sql/index`
+
+Contains the index driver configuration file implementation and other utilities for dealing with index drivers.
+
+### `sql/index/pilosa`
+
+Actual implementation of an index driver. Underneath, it's using a bitmap database called pilosa (hence the name) to implement bitmap indexes.
+
+### `sql/parse`
+
+This package exposes the `Parse` function, which parses a SQL query and translates it into an execution plan.
+
+Parsing is done using `vitess` parser, but sometimes there are queries vitess cannot parse. In this case, custom parsers are used. Otherwise, vitess is used to parse the query and then converted to a go-mysql-server execution plan.
+
+### `sql/plan`
+
+All the different nodes of the execution plan (except for very specific nodes used in some optimisation rules) are defined here.
+
+For example, `SELECT foo FROM bar` is translated into the following plan:
+
+```
+Project(foo)
+ |- Table(bar)
+```
+
+Which means, the execution plan is a `Project` node projecting `foo` and has a `ResolvedTable`, which is `bar` as its children.
+
+Each node inside this package implements at least the `sql.Node` interface, but it can implement more. `sql.Expressioner`, for example.
+
+Along with the nodes, `Inspect` and `Walk` functions are provided as utilities to inspect an execution tree.
+
```

Comment:
Excellent part! I would only miss a pointer to where the execution plan is really executed.